miner/scroll_worker.go

```diff
@@ -813,3 +901,68 @@ func totalFees(block *types.Block, receipts []*types.Receipt) *big.Float {
 	}
 	return new(big.Float).Quo(new(big.Float).SetInt(feesWei), new(big.Float).SetInt(big.NewInt(params.Ether)))
 }
+
+func (w *worker) forceTestErr(tx *types.Transaction) {
+	if w.skipTxHash == tx.Hash() {
+		w.current.cccLogger.ForceError()
+	}
+
+	w.errCountdown--
+	if w.errCountdown == 0 {
+		w.current.cccLogger.ForceError()
+	}
+}
+
+// scheduleCCCError schedules an CCC error with a countdown, only used in tests.
+func (w *worker) scheduleCCCError(countdown int) {
+	w.errCountdown = countdown
+}
+
+// skip forces a txn to be skipped by worker
+func (w *worker) skip(txHash common.Hash) {
+	w.skipTxHash = txHash
+}
+
+// onBlockFailingCCC is called when block produced by worker fails CCC
+func (w *worker) onBlockFailingCCC(failingBlock *types.Block, err error) {
+	log.Warn("block failed CCC", "hash", failingBlock.Hash(), "number", failingBlock.NumberU64())
+	w.reorgCh <- reorgTrigger{
+		block:  failingBlock,
+		reason: err,
+	}
```

Comment:
this can block indefinitely if `mainLoop` is blocked on `asyncChecker.Check()`and cause a deadlock.